### PR TITLE
dind: Ensure unique node config and fix perms

### LINKED ISSUE #11274: networking test flake "certificate is valid for nettest-node-1, 172.17.0.3, not nettest-node-2"
Seen in https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/6833:

`Oct  7 17:38:40.311: INFO: Error running &{/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc [oc logs --namespace=extended-test-get-flows-57ect-zuw9v --config=/tmp/extended-test-get-flows-57ect-zuw9v-user.kubeconfig flow-checklhsi6] []   Error from server: Get https://nettest-node-2:10250/containerLogs/extended-test-get-flows-57ect-zuw9v/flow-checklhsi6/flow-check: x509: certificate is valid for nettest-node-1, 172.17.0.3, not nettest-node-2`

This is the same symptom as #11025 but presumably not the same cause. This was in a post-#11061 tree, so it may actually be caused by a dind deployment bug this time...

@marun

### PR DESCRIPTION
**Pull request by marun**
A dind deployment flake on #11315 revealed that config permissions are not being set correctly to allow cluster artifacts to be saved.  This change ensures the cluster configuration is readable so it can be saved.

cc: @openshift/networking @stevekuznetsov 

Should also fix #11274

### CODE REVIEW

**Comment by marun** [comment]
[test]

**Comment by marun** [comment]
This fixes more than just permissions.  Previously node configuration was always generated into the same path on the shared volume - ``/data/openshift.local.config`` - instead of a node-specific path  - ``/data/openshift.local.config/node-*``.  This was likely resulting in race conditions where one or more nodes could end up using the same configuration.

File: images/dind/master/openshift-generate-master-config.sh [hunk 1]
```diff
@@ -32,9 +32,11 @@ function ensure-master-config() {
     --master="https://${ip_addr}:8443" \
     --network-plugin="${OPENSHIFT_NETWORK_PLUGIN}"
 
-  # ensure the configuration is readable outside of the container
-  find "${config_path}" -exec chmod ga+rw {} \;
-  find "${config_path}" -type d -exec chmod ga+x {} \;
+  # ensure the configuration can be used outside of the container
+  chmod ga+rx "${config_path}"
```
Comment (stevekuznetsov) [inline images/dind/master/openshift-generate-master-config.sh]: Can we collapse this:
```sh
chmod -R ga+rX "${config_path}"
chmod ga+w "${master_path}/admin.kubeconfig"
```

**Comment by stevekuznetsov** [comment]
`chmod`s look good. Do we want to merge both this one and #11298 and see which goes in first? Or close that other and one focus on this one?

**Comment by marun** [comment]
@stevekuznetsov Ok, I'll close the other one. I tend to try to split things up but these fixes are very closely related.

**Comment by danwinship** [comment]
lgtm i think. let's see what the tests say...

**Comment by marun** [comment]
flake #8571

re-[test]

**Comment by marun** [comment]
@danwinship well, the check job passed.  shame about the others that have nothing to do with this PR.

**Comment by marun** [comment]
flake #8571

re-[test]

**Comment by marun** [comment]
@openshift/networking The flakes are unrelated, please merge.

**Comment by marun** [comment]
flake #8571 re-[test]

**Comment by knobunc** [comment]
[merge]

**Comment by marun** [comment]
holy merge queue, batman

**Comment by knobunc** [comment]
Flake https://github.com/openshift/origin/issues/10663 [merge]

**Comment by stevekuznetsov** [comment]
@knobunc the test job is running and failing, the merge has not yet occurred. I don't *think* re-triggering a merge will shove you to the end of the line, but it may be ... no need to re-tag

**Comment by knobunc** [comment]
@stevekuznetsov Thanks, I wasn't sure if the test fail was going to prevent the merge.

**Comment by stevekuznetsov** [comment]
> the test fail was going to prevent the merge.

today the relationship is one-way in that successful tests (against a master that hasn't changed) can help you, but failed tests can never hurt you

**Comment by marun** [comment]
[test] curious to see the networking job runtime

**Comment by dcbw** [comment]
lgtm

**Pull request merged**